File: frontend/src/models/gameRule.ts
export type TBasicGameRule = {
  divisibleNumber: number;
  replacedWord: string;
};

export type TGameRule = TBasicGameRule & {
  ruleId: string;
  gameId: string;
};

export type TUpdateGameRulePayload = Partial<TBasicGameRule> & {
  ruleId: string;
};

export type TAddGameRulePayload = TBasicGameRule & {
  gameId: string;
};

File: backend/FinalAssignmentBE/Models/GameRule.cs
using System.ComponentModel.DataAnnotations;

namespace FinalAssignmentBE.Models;

public class GameRule
{
    [Key]
    public long RuleId { get; set; }
    public int DivisibleNumber { get; set; }
    public string ReplacedWord { get; set; }
    public long GameId { get; set; }
    public Game Game { get; set; }
}